clang/lib/CodeGen/CGExprConstant.cpp

```diff
@@ -1126,6 +1126,7 @@ class ConstExprEmitter :
     case CK_NonAtomicToAtomic:
     case CK_NoOp:
     case CK_ConstructorConversion:
+    case CK_HLSLArrayRValue:
```

Comment:
We shouldn't accept CK_HLSLArrayRValue casts here (or if we do, we need to do something different with them).